File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int insert_ptr=1;
        int frequency=1;
        for (int i=1;i<nums.length;i++){
            if(nums[i]==nums[i-1]){
                frequency++;
            }else{
                frequency=1;
            }
            if (frequency<=2){
                nums[insert_ptr++]=nums[i];
                // insert_ptr++;
            }
        }
        return insert_ptr;
    }
}

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        insert_index = 1
        index = 1
        repetition = 1
        while index < len(nums):
            if nums[index] == nums[index - 1]:
                repetition += 1
            else:
                repetition = 1

            if repetition <= 2:
                nums[insert_index] = nums[index]
                insert_index += 1
            index += 1
        return insert_index
